yarn-project/end-to-end/src/e2e_p2p/slashing.test.ts

```diff
@@ -0,0 +1,278 @@
+    }
+
+    const rollup = getContract({
+      address: t.ctx.deployL1ContractsValues!.l1ContractAddresses.rollupAddress.toString(),
+      abi: RollupAbi,
+      client: t.ctx.deployL1ContractsValues!.walletClient,
+    });
+
+    const slasherContract = getContract({
+      address: getAddress(await rollup.read.SLASHER()),
+      abi: SlasherAbi,
+      client: t.ctx.deployL1ContractsValues.publicClient,
+    });
+
+    const slashingProposer = getContract({
+      address: getAddress(await slasherContract.read.PROPOSER()),
+      abi: SlashingProposerAbi,
+      client: t.ctx.deployL1ContractsValues.publicClient,
+    });
+
+    const slashFactory = getContract({
+      address: getAddress(t.ctx.deployL1ContractsValues.l1ContractAddresses.slashFactoryAddress.toString()),
+      abi: SlashFactoryAbi,
+      client: t.ctx.deployL1ContractsValues.publicClient,
+    });
+
+    const slashingInfo = async () => {
+      const bn = await t.ctx.cheatCodes.eth.blockNumber();
+      const slotNumber = await rollup.read.getCurrentSlot();
+      const roundNumber = await slashingProposer.read.computeRound([slotNumber]);
+      const instanceAddress = t.ctx.deployL1ContractsValues.l1ContractAddresses.rollupAddress.toString();
+      const info = await slashingProposer.read.rounds([instanceAddress, roundNumber]);
+      const leaderVotes = await slashingProposer.read.yeaCount([instanceAddress, roundNumber, info[1]]);
+      return { bn, slotNumber, roundNumber, info, leaderVotes };
+    };
+
+    const waitForL1Block = async () => {
+      // Send and wait an l1 block
+      await t.ctx.deployL1ContractsValues.publicClient.waitForTransactionReceipt({
+        hash: await t.ctx.deployL1ContractsValues.walletClient.sendTransaction({
+          to: t.ctx.deployL1ContractsValues.walletClient.account.address,
+          value: 1n,
+          account: t.ctx.deployL1ContractsValues.walletClient.account,
+        }),
+      });
+    };
+
+    t.ctx.aztecNodeConfig.validatorReexecute = false;
+
+    // create our network of nodes and submit txs into each of them
+    // the number of txs per node and the number of txs per rollup
+    // should be set so that the only way for rollups to be built
+    // is if the txs are successfully gossiped around the nodes.
+    t.logger.info('Creating nodes');
+    nodes = await createNodes(
+      t.ctx.aztecNodeConfig,
+      t.ctx.dateProvider,
+      t.bootstrapNodeEnr,
+      NUM_NODES,
+      BOOT_NODE_UDP_PORT,
+      DATA_DIR,
+      // To collect metrics - run in aztec-packages `docker compose --profile metrics up` and set COLLECT_METRICS=true
+      shouldCollectMetrics(),
+    );
+
+    // We are overriding the slashing amount to 1, such that the slashing will "really" happen.
+    for (const node of nodes) {
+      const seqClient = node.getSequencer();
+      if (!seqClient) {
+        throw new Error('Sequencer not found');
+      }
+      const sequencer = (seqClient as any).sequencer;
```

Comment:
is this line needed

Reply:
Probably not needed as separate, can be smashed together.